src/Imath/half.h

```diff
@@ -17,33 +17,149 @@
+///
+/// **Representation of a 16-bit half:**
+///
+/// Here is the bit-layout for a half number, h:
+///
+///     15 (msb)
+///     |
+///     | 14  10
+///     | |   |
+///     | |   | 9        0 (lsb)
+///     | |   | |        |
+///     X XXXXX XXXXXXXXXX
+///
+///     s e     m
+///
+/// S is the sign-bit, e is the exponent and m is the significand.
+///
+/// If e is between 1 and 30, h is a normalized number:
+///
+///             s    e-15
+///     h = (-1)  * 2     * 1.m
+///
+/// If e is 0, and m is not zero, h is a denormalized number:
+///
+///             S    -14
+///     h = (-1)  * 2     * 0.m
+///
+/// If e and m are both zero, h is zero:
+///
+///     h = 0.0
+///
+/// If e is 31, h is an "infinity" or "not a number" (NAN),
+/// depending on whether m is zero or not.
+///
+/// Examples:
+///
+///     0 00000 0000000000 = 0.0
+///     0 01110 0000000000 = 0.5
+///     0 01111 0000000000 = 1.0
+///     0 10000 0000000000 = 2.0
+///     0 10000 1000000000 = 3.0
+///     1 10101 1111000001 = -124.0625
+///     0 11111 0000000000 = +infinity
+///     1 11111 0000000000 = -infinity
+///     0 11111 1000000000 = NAN
+///     1 11111 1111111111 = NAN
+///
+/// **Conversion via Lookup Table:**
+///
+/// Converting from a float to a half requires some non-trivial bit
+/// manipulations.  In some cases, this makes conversion relatively
+/// slow, but the most common case is accelerated via table lookups.
+///
+/// Converting back from a half to a float is easier because we don't
+/// have to do any rounding.  In addition, there are only 65536
+/// different half numbers; we can convert each of those numbers once
+/// and store the results in a table.  Later, all conversions can be
+/// done using only simple table lookups. (Although this is
+/// arguably the fastest way to do this, and Imath has an implementation in
+/// hardware now)
+///
+/// **Conversion via Hardware:**
+///
+/// For Imath v3.1, the conversion routines have been extended to use
+/// hardware instructions when enabled by the compiler.
+///
+/// Furthermore, additional control is possible for embedded
 /// systems. First, the library can be configured to disable
 /// generation of the half to float table. This table is faster on x86
 /// hardware, but for embedded systems may artificially bloat the
 /// executable. Without this, a smaller table is used for
 /// acceleration. An implementation can add a preprocessor #define
 /// IMATH_HALF_NO_TABLES_AT_ALL, which will eliminate all tables.
 ///
+/// **Conversion Performance Comparison:**
+///
 /// Testing on a Core i9, the timings are approximately:
 ///
-/// half to float table: 0.71 ns / call
-/// no table: 1.06 ns / call
-/// f16c: 0.45 ns / call
+/// - half to float table: 0.71 ns / call
+/// - no table: 1.06 ns / call
+/// - f16c: 0.45 ns / call
 ///
-/// original: 5.2 ns / call
-/// no exp table + opt: 1.27 ns / call
-/// f16c: 0.45 ns / call
+/// - original: 5.2 ns / call
```

Comment:
This should have a header on it to indicate it is for float to half conversion, thanks! (I missed that originally, appreciate you touching all this up)